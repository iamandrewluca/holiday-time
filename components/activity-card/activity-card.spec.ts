import { describe, it, expect } from '@jest/globals'
import { render } from '@testing-library/vue'
import ActivityCard from './activity-card.component.vue'

describe('ActivityCard', () => {
  it('should render and match snapshot', () => {
    const { baseElement } = render(ActivityCard, {
      props: {
        title: 'Product title',
        description: 'Product description',
        image: 'https:/picsum.photos/seed/123/1920/1080',
        price: '$ 12.00',
      },
    })

    expect(baseElement).toMatchSnapshot()
  })
})
