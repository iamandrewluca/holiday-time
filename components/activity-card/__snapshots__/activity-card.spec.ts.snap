// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard should render and match snapshot 1`] = `
<body>
  <div>
    <div
      class="activity-card"
    >
      <div
        class="root"
      >
        <div
          class="inner"
        >
          <button
            class="activity-card__star"
            data-testid="add-to-wishlist"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
           
          <img
            alt=""
            class="activity-card__image"
            src="https:/picsum.photos/seed/123/1920/1080"
          />
           
          <span
            class="activity-card__price"
          >
            $ 12.00
          </span>
        </div>
      </div>
       
      <h3
        class="activity-card__title"
      >
        Product title
      </h3>
       
      <p
        class="activity-card__description"
      >
        Product description
      </p>
       
      <button
        class="w-full"
        data-testid="add-to-cart"
      >
        Add to Cart
      </button>
    </div>
  </div>
</body>
`;
